### PR TITLE
Istio 1.3.1 support

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Add support for Istio 1.3.1

### CODE REVIEW

File: pkg/resources/nodeagent/daemonset.go [hunk 1]
```diff
@@ -75,6 +75,10 @@ func (r *Reconciler) daemonSet() runtime.Object {
 									Name:  "Trust_Domain",
```
Comment (Laci21) [inline pkg/resources/nodeagent/daemonset.go]: Why are you keeping this one?

File: pkg/apis/istio/v1beta1/defaults.go [hunk 2]
```diff
@@ -140,6 +140,12 @@ func SetDefaults(config *Istio) {
 	if config.Spec.Pilot.EnableProtocolSniffing == nil {
 		config.Spec.Pilot.EnableProtocolSniffing = util.BoolPointer(true)
 	}
+	if config.Spec.Pilot.EnableProtocolSniffingOutbound == nil {
+		config.Spec.Pilot.EnableProtocolSniffingOutbound = config.Spec.Pilot.EnableProtocolSniffing
+	}
+	if config.Spec.Pilot.EnableProtocolSniffingInbound == nil {
+		config.Spec.Pilot.EnableProtocolSniffingInbound = config.Spec.Pilot.EnableProtocolSniffing
```
Comment (Laci21) [inline pkg/apis/istio/v1beta1/defaults.go]: This seems to be `false` by default: https://github.com/istio/istio/blob/6d0953f791f76f614c2478f463eef166ea399105/pilot/pkg/features/pilot.go#L183

Also, we might consider adding the description from the link to the variables like: `If enabled, protocol sniffing will be used for inbound listeners whose port protocol is not specified or unsupported`

File: pkg/resources/pilot/deployment.go [hunk 3]
```diff
@@ -157,6 +157,20 @@ func (r *Reconciler) containers() []apiv1.Container {
 		})
 	}
 
+	if util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffingOutbound) {
+		discoveryContainer.Env = append(discoveryContainer.Env, apiv1.EnvVar{
+			Name:  "PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_OUTBOUND",
+			Value: "true",
```
Comment (Laci21) [inline pkg/resources/pilot/deployment.go]: Are u sure we always want to set this to `true`? If `util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffingOutbound)` is `false` then we don't set this env var, but won't it be initialized to be `true` anyway: https://github.com/istio/istio/blob/6d0953f791f76f614c2478f463eef166ea399105/pilot/pkg/features/pilot.go#L177?

File: pkg/resources/nodeagent/daemonset.go (hunk 1 above)
Comment (waynz0r) [reply pkg/resources/nodeagent/daemonset.go]: I left it there so the operator to be backward compatible.

File: pkg/apis/istio/v1beta1/defaults.go (hunk 2 above)
Comment (waynz0r) [reply pkg/apis/istio/v1beta1/defaults.go]: 👍

File: pkg/resources/pilot/deployment.go (hunk 3 above)
Comment (waynz0r) [reply pkg/resources/pilot/deployment.go]: 👍

File: pkg/resources/pilot/deployment.go [hunk 4]
```diff
@@ -150,12 +150,21 @@ func (r *Reconciler) containers() []apiv1.Container {
...
-	if util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffing) {
-		discoveryContainer.Env = append(discoveryContainer.Env, apiv1.EnvVar{
-			Name:  "PILOT_ENABLE_PROTOCOL_SNIFFING",
-			Value: "true",
-		})
-	}
+	// TODO - remove it in the next minor -- left it here for backward compatibility
+	discoveryContainer.Env = append(discoveryContainer.Env, apiv1.EnvVar{
+		Name:  "PILOT_ENABLE_PROTOCOL_SNIFFING",
+		Value: strconv.FormatBool(util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffingOutbound)),
```
Comment (Laci21) [inline pkg/resources/pilot/deployment.go]:
```suggestion
		Value: strconv.FormatBool(util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffing)),
```

**Pull request merged**